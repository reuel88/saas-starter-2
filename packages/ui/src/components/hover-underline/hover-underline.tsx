import { ReactNode } from "react";
import { cn } from "@repo/utilities";

type HoverUnderlineProps = {
  children: ReactNode;
  className?: string;
};
export const HoverUnderline = ({
  children,
  className = "",
}: HoverUnderlineProps) => {
  return (
    <span className="group">
      {children}
      <span
        className={cn(
          "block h-1 w-0 bg-hue-500 transition-all group-hover:w-full",
          className,
        )}
        aria-hidden="true"
      />
    </span>
  );
};
