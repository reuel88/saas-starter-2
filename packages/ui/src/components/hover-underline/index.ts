export { HoverUnderline } from "./hover-underline";
