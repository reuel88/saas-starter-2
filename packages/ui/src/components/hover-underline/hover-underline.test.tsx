import { render, screen } from "@testing-library/react";
import { HoverUnderline } from "./hover-underline";

describe("HoverUnderline", () => {
  it("should render correctly", () => {
    // Arrange
    const Content = () => <span>Test</span>;

    // Act
    const { baseElement } = render(
      <HoverUnderline>
        <Content />
      </HoverUnderline>,
    );
    const contentText = screen.getByText("Test");

    // Assert
    expect(contentText).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
