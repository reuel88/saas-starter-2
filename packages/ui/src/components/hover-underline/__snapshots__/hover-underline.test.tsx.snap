// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HoverUnderline > should render correctly 1`] = `
<body>
  <div>
    <span
      class="group"
    >
      <span>
        Test
      </span>
      <span
        aria-hidden="true"
        class="block h-1 w-0 bg-hue-500 transition-all group-hover:w-full"
      />
    </span>
  </div>
</body>
`;
